test/raft/unit/test_snapshot.c

```diff
@@ -407,79 +524,242 @@ static void pool_rpc_to_expired(struct rpc *rpc)
 	progress();
 }
 
-/* TODO(alberto): combine them with tests above once the rest is in place.
- * Dispatch to one of two implementations ut or pool in general functions. */
-TEST(snapshot_leader, pool_timeouts, set_up, tear_down, 0, NULL) {
+static bool pool_is_main_thread_op(void) {
+	return thread_identifier == MAGIC_MAIN_THREAD;
+}
+
+static void pool_ht_create_op(pool_work_t *w)
+{
+	if (global_fixture.is_leader) {
+		PRE(!global_fixture.leader.ops->is_main_thread());
+	} else {
+		PRE(!global_fixture.follower.ops->is_main_thread());
+	}
+	(void)w;
+}
+
+static void pool_fill_ht_op(pool_work_t *w)
+{
+	if (global_fixture.is_leader) {
+		PRE(!global_fixture.leader.ops->is_main_thread());
+	} else {
+		PRE(!global_fixture.follower.ops->is_main_thread());
+	}
+	(void)w;
+}
+
+static void pool_write_chunk_op(pool_work_t *w)
+{
+	struct work *work = CONTAINER_OF(w, struct work, pool_work);
+	struct follower *follower = CONTAINER_OF(work, struct follower, work);
+	PRE(!follower->ops->is_main_thread());
+}
+
+static void pool_read_sig_op(pool_work_t *w)
+{
+	struct work *work = CONTAINER_OF(w, struct work, pool_work);
+	struct follower *follower = CONTAINER_OF(work, struct follower, work);
+	PRE(!follower->ops->is_main_thread());
+}
+
+struct uv_sender_send_data {
+	struct sender *s;
+	sender_cb_op cb;
+};
+
+void uv_sender_send_cb(uv_work_t *req) {
+	(void)req;
+}
+
+void uv_sender_send_after_cb(uv_work_t *req, int status) {
+	(void)req;
+	(void)status;
+	global_fixture.msg_valid = true;
+	struct uv_sender_send_data *data = req->data;
+	data->cb(data->s, 0);
+}
+
+int uv_sender_send_op(struct sender *s,
```

Comment:
If I'm not mistaken, `uv_sender_*()` functions can be `static`.

Reply:
I always forget `static` and `const`, always.